bindings/pydeck-carto/pydeck_carto/carto_auth/layer_validator.py

```diff
@@ -0,0 +1,64 @@
+from pydeck import Layer, types
+from pydeck_carto.carto_auth.auth import CartoAuth
+from pydeck_carto.layer import MapType, CartoConnection, GeoColumnType
+
+
+def validate_attr(layer, attr, param_type=types.String):
+    if not hasattr(layer, attr):
+        raise ValueError(f'Missing parameter {attr} on the layer')
+    attr_value = getattr(layer, attr)
+    if not attr_value:
+        raise ValueError(f'Parameter {attr} not valid')
+    if not isinstance(attr_value, param_type):
+        raise ValueError(f'Parameter {attr} must be a {param_type}')
+
+
+def is_valid_carto_layer(layer: Layer, carto_auth: CartoAuth):
+    if layer.type != 'CartoLayer':
+        raise ValueError('The layer type must be "CartoLayer"')
+
+    validate_attr(layer, 'credentials', dict)
+    validate_attr(layer, 'data', str)
+
+    required_strings = ['type_', 'connection']
+    for attr in required_strings:
+        validate_attr(layer, attr)
+
+    if layer.type_ not in [MapType.QUERY, MapType.TABLE, MapType.TILESET]:
+        raise ValueError(f'Invalid layer type selected')
+
+    for attr in ('accessToken', 'apiVersion', 'apiBaseUrl'):
+        if attr not in layer.credentials:
+            raise ValueError(f'Missing {attr} from layer.credentials')
+
+    if 'carto.com' not in layer.credentials['apiBaseUrl']:
```

Comment:
Remove this check. CARTO Self-hosted can have a different apiBaseUrl without carto.com.

Reply:
Ok